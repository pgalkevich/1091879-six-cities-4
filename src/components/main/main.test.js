import React from "react";
import renderer from "react-test-renderer";
import Main from "./main";

const offers = [
  {
    bedrooms: 5,
    city: {
      name: `Paris`,
      location: {
        latitude: 48.8566,
        longitude: 2.3522,
        zoom: 16
      }
    },
    description: `Some description`,
    goods: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    host: {
      avatar: `img/1.jpg`,
      id: 1,
      is_pro: false,
      name: `Name`
    },
    id: 1,
    images: [
      `room.jpg`,
      `apartment-01.jpg`,
      `apartment-02.jpg`,
      `apartment-03.jpg`,
      `studio-01.jpg`,
      `apartment-01.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 48.8566,
      longitude: 2.3522,
      zoom: 10
    },
    max_adults: 5,
    preview_image: `img/apartment-01.jpg`,
    price: 10,
    rating: 5,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
  },
  {
    bedrooms: 5,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.9375,
        longitude: 6.9603,
        zoom: 16
      }
    },
    description: `Some description`,
    goods: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    host: {
      avatar: `img/1.jpg`,
      id: 1,
      is_pro: false,
      name: `Name`
    },
    id: 1,
    images: [
      `room.jpg`,
      `apartment-01.jpg`,
      `apartment-02.jpg`,
      `apartment-03.jpg`,
      `studio-01.jpg`,
      `apartment-01.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 48.8566,
      longitude: 2.3522,
      zoom: 10
    },
    max_adults: 5,
    preview_image: `img/apartment-01.jpg`,
    price: 10,
    rating: 5,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
  }
];
const cities = [
  {
    name: `Amsterdam`,
    location: {
      longitude: 10,
      latitude: 10,
      zoom: 16
    }
  },
  {
    name: `Hamburg`,
    location: {
      longitude: 10,
      latitude: 10,
      zoom: 16
    }
  },
  {
    name: `Paris`,
    location: {
      longitude: 10,
      latitude: 10,
      zoom: 16
    }
  },
  {
    name: `Dusseldorf`,
    location: {
      longitude: 10,
      latitude: 10,
      zoom: 16
    }
  },
  {
    name: `Brussels`,
    location: {
      longitude: 10,
      latitude: 10,
      zoom: 16
    }
  },
  {
    name: `Cologne`,
    location: {
      longitude: 10,
      latitude: 10,
      zoom: 16
    }
  },
];
const currentCity =  {
  name: `Amsterdam`,
  location: {
    longitude: 10,
    latitude: 10,
    zoom: 16
  }
};


it(`Should Main render correctly`, () => {
  const tree = renderer
    .create(
        <Main
          currentCity={currentCity}
          offers={offers}
          onCardHover={() => {}}
          onTitleClick={() => {}}
          onMenuItemClick={() => {}}
          cities={cities}
        />,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
