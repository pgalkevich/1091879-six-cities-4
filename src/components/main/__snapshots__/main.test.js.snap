// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Main render correctly 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          0
           places to stay in 
          Amsterdam
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            className="places__sorting-type"
            tabIndex="0"
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom places__options--opened"
          >
            <li
              className="places__option places__option--active"
              tabIndex="0"
            >
              Popular
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: low to high
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: high to low
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Top rated first
            </li>
          </ul>
        </form>
        <div
          className="cities__places-list places__list tabs__content"
        />
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;
